composer/trainer/dist_strategy.py

```diff
@@ -261,21 +249,13 @@ def sync_hook(*args):
             raise RuntimeError('CUDA out of memory encountered on a different rank')
 
     kwargs = {}
-    if is_torch_2_0:
-        # Support of new parameter `use_orig_params` in PyTorch 2.0 or higher.
-        # Setting this to `True` has FSDP use `module`'s original parameters via method
-        # `nn.Module.named_parameters` instead of FSDP's internal class `FlatParameter`. However,
-        # setting it to `False` exposes FSDP's internal class `FlatParameter` via method
-        # `nn.Module.named_parameters`.
-        # Setting it to `True` is mandatory when using `torch.compile()`.
-        kwargs['use_orig_params'] = fsdp_config['use_orig_params']
-        if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0'):
-            if 'device_mesh' in fsdp_config:
-                from torch.distributed._tensor import init_device_mesh
-                kwargs['device_mesh'] = init_device_mesh(
-                    'cuda',
-                    tuple([int(x) for x in fsdp_config['device_mesh']]),
-                )
+    if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0'):
```

Comment:
Can we do this everywhere instead of the 2.2.9 thing?

Reply:
Let's clean up in a separate PR? I think it's a different issue (but agree)

Reply:
sure, jira plz